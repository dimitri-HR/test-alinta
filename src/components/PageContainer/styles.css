.pageContainer {
    margin: 0 auto;
    max-width: 104rem;
    padding: 1.6rem;
}

@media only screen and (min-width: 500px) {
    .pageContainer {
        padding: 3.2rem;
    }
}

@media only screen and (min-width: 768px) {
    .pageContainer {
        padding: 6.4rem;
    }
}
