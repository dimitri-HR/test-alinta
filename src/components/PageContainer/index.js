import React from 'react'
import { string, node } from 'prop-types'
import cx from 'classnames'

import styles from './styles.css'

// Wraps components to provide consistent styling
// Pass in the 'className' prop to add extra styles
const PageContainer = ({ className, children }) => (
  <div className={cx(styles.pageContainer, className)}>
    {children}
  </div>
)

PageContainer.propTypes = {
  className: string,
  children: node,
}

PageContainer.defaultProps = {
  className: '',
  children: null,
}

export default PageContainer
