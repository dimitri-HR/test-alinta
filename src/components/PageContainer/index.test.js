import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import PageContainer from '.'

describe('PageContainer', () => {
  it('renders without errors', () => {
      const component = renderer.create(
      <PageContainer>
        <div>Children</div>
      </PageContainer>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('can pass correct className', () => {
      const component = shallow(
      <PageContainer className='testClass'>
        <div>Children</div>
      </PageContainer>
    )
    expect(component.find('div').first().hasClass('testClass')).toBe(true)
  })
})
