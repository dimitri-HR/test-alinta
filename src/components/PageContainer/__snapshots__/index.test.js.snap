// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer renders without errors 1`] = `
<div
  className="pageContainer"
>
  <div>
    Children
  </div>
</div>
`;
